sdk/core/src/main/java/com/klaviyo/core/networking/KlaviyoNetworkMonitor.kt

```diff
@@ -80,7 +80,7 @@ internal object KlaviyoNetworkMonitor : NetworkMonitor {
      */
     override fun isNetworkConnected(): Boolean = connectivityManager
         .getNetworkCapabilities(connectivityManager.activeNetwork)
-        ?.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) ?: false
+        ?.hasCapability(NetworkCapabilities.NET_CAPABILITY_VALIDATED) ?: false
```

Comment:
Related change -- we were only checking if the network can connect to the internet in theory, not a validated check. This should reduce wasted attempts if the system already knows that it isn't fully online.